mithril-signer/src/database/repository/protocol_initializer_repository.rs

```diff
@@ -0,0 +1,308 @@
+            )
+            .await
+            .unwrap();
+
+        assert!(res.is_some());
+        assert_eq!(
+            protocol_initializers[0].1.get_stake(),
+            res.unwrap().get_stake()
+        );
+    }
+
+    #[tokio::test]
+    async fn get_protocol_initializer_for_empty_epoch() {
+        let store = init_store(2, None).await;
+        let res = store.get_protocol_initializer(Epoch(0)).await.unwrap();
+
+        assert!(res.is_none());
+    }
+
+    #[tokio::test]
+    async fn get_protocol_initializer_for_existing_epoch() {
+        let store = init_store(2, None).await;
+        let res = store.get_protocol_initializer(Epoch(1)).await.unwrap();
+
+        assert!(res.is_some());
+    }
+
+    #[tokio::test]
+    async fn get_last_protocol_initializer_return_last_one_first() {
+        let store = init_store(0, None).await;
+        let values = setup_protocol_initializers(10);
+        store_protocol_initializers(&store, &values).await;
+
+        let res = store.get_last_protocol_initializer(3).await.unwrap();
+
+        assert_eq!(3, res.len());
+        assert_eq!(values[9].0, res[0].0);
+        assert_eq!(values[8].0, res[1].0);
+        assert_eq!(values[7].0, res[2].0);
+    }
+
+    #[tokio::test]
+    async fn get_last_protocol_initializer_return_all_when_too_few_records() {
+        let store = init_store(0, None).await;
+        let values = setup_protocol_initializers(2);
+        store_protocol_initializers(&store, &values).await;
+
+        let res = store.get_last_protocol_initializer(3).await.unwrap();
+
+        assert_eq!(2, res.len());
+        assert_eq!(values[1].0, res[0].0);
+        assert_eq!(values[0].0, res[1].0);
+    }
+
+    #[tokio::test]
+    async fn prune_epoch_older_than_threshold() {
+        const PROTOCOL_INITIALIZER_PRUNE_EPOCH_THRESHOLD: u64 = 10;
+
+        let store = init_store(0, Some(PROTOCOL_INITIALIZER_PRUNE_EPOCH_THRESHOLD)).await;
+        let values = setup_protocol_initializers(2);
+        store_protocol_initializers(&store, &values).await;
+
+        store
+            .prune(Epoch(2) + PROTOCOL_INITIALIZER_PRUNE_EPOCH_THRESHOLD)
+            .await
+            .unwrap();
+
+        let res = store.get_last_protocol_initializer(10).await.unwrap();
+
+        assert_eq!(1, res.len());
+        assert_eq!(Epoch(2), res[0].0);
+    }
+
+    #[tokio::test]
+    async fn without_threshold_nothing_is_pruned() {
+        let store = init_store(0, None).await;
+        let values = setup_protocol_initializers(2);
+        store_protocol_initializers(&store, &values).await;
+
+        store.prune(Epoch(100)).await.unwrap();
+
+        let res = store.get_last_protocol_initializer(10).await.unwrap();
+        assert_eq!(2, res.len());
+    }
+
+    #[tokio::test]
+    async fn should_migrate_data_from_adapter() {
+        let migrations = crate::database::migration::get_migrations();
+
+        // TODO: Do it in test_helper (it is done by build_main_db_connection)
```

Comment:
Those remaining TODOs for `create_connection_builder` will be addressed in a future PR, is that correct?

Reply:
Done